### PR TITLE
Change bitvector from 1024 to 512

### PR DESCRIPTION
**Pull request by terencechain**
Aligning to latest Altair [spec](https://github.com/ethereum/eth2.0-specs/releases/tag/v1.1.0-alpha.4). Update sync commiteee's bitvector from 1024 to 512

### CODE REVIEW

File: bitvector512.go [hunk 1]
```diff
@@ -5,26 +5,26 @@ import (
...
 	return byteArray[:]
 }
 
 // BitAt returns the bit value at the given index. If the index requested
 // exceeds the number of bits in the bitvector, then this method returns false.
-func (b Bitvector1024) BitAt(idx uint64) bool {
+func (b Bitvector512) BitAt(idx uint64) bool {
 	// Out of bounds, must be false.
-	if idx >= b.Len() {
+	if idx >= b.Len() || idx >= 512 {
```
Comment (prestonvanloon) [inline bitvector512.go:27]: This is not necessary, Len is always 512

File: bitvector512.go [hunk 2]
```diff
@@ -35,9 +35,9 @@ func (b Bitvector1024) BitAt(idx uint64) bool {
 // SetBitAt will set the bit at the given index to the given value. If the index
 // requested exceeds the number of bits in the bitvector, then this method returns
 // false.
-func (b Bitvector1024) SetBitAt(idx uint64, val bool) {
+func (b Bitvector512) SetBitAt(idx uint64, val bool) {
 	// Out of bounds, do nothing.
-	if idx >= b.Len() {
+	if idx >= b.Len() || idx >= 512 {
```
Comment (prestonvanloon) [inline bitvector512.go:40]: This is not necessary, Len is always 512

File: bitvector512.go [hunk 3]
```diff
@@ -5,26 +5,26 @@ import (
...
 
-const bitvector1024ByteSize = 128
-const bitvector1024BitSize = bitvector1024ByteSize * 8
+const bitvector512ByteSize = 64
+const bitvector512BitSize = bitvector512ByteSize * 8
 
-// NewBitvector1024 creates a new bitvector of size 1024.
-func NewBitvector1024() Bitvector1024 {
-	byteArray := [bitvector1024ByteSize]byte{}
+// NewBitvector512 creates a new bitvector of size 1024.
```
Comment (prestonvanloon) [inline bitvector512.go:17]:
```suggestion
// NewBitvector512 creates a new bitvector of size 512.
```

File: bitvector512_test.go [hunk 4]
```diff
@@ -0,0 +1,443 @@
...
+				tt.bitlist,
+				tt.idx,
+				tt.bitlist.BitAt(tt.idx),
+				tt.want,
+			)
+		}
+	}
+}
+
+func TestBitvector512_SetBitAt(t *testing.T) {
```
Comment (prestonvanloon) [inline bitvector512_test.go:98]: Please add an "out of bounds" test where idx is greater than or equal to 512

**Review by rauljordan — APPROVED** [review]

**Review by prestonvanloon — APPROVED** [review]

**Pull request merged**